pkgs/tools/networking/tridactyl-native/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tridactyl-native";
+  # this is actually the version of tridactyl itself; the native messenger will
+  # probably not change with every tridactyl version
+  version = "1.14.9";
+
+  src = fetchFromGitHub {
+    owner = "tridactyl";
+    repo = "tridactyl";
+    rev = version;
+    sha256 = "0d80c744qfv6jd03cmdp3p71xaj8lq8jzsa2m24jxv9q4ks2dcmj";
+  };
+  sourceRoot = "source/native";
+
+  buildPhase = ''
+    sed -i -e "s|REPLACE_ME_WITH_SED|$out/native_main.py|" "tridactyl.json"
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/lib/mozilla/native-messaging-hosts"
+    cp tridactyl.json "$out/lib/mozilla/native-messaging-hosts/"
+    cp native_main.py "$out/"
```

Comment:
You need to wrap this with the correct python interpreter and deps and then installing to $out/bin/tridactyl makes more sense and also prevents potential collisions. Something like
```nix
{ python3Packages }: /* ... */ {
  # ...
  nativeBuildInputs = [ python3Packages.wrapPython ];
  pythonPath = with python3Packages; [ ... ];
  installPhase = ''
    # ...
    cp native_main.py $out/bin/tridactyl
    wrapPythonPrograms
  '';
}
```

should do it. (No idea if it needs any deps at all though, haven't checked).

Reply:
Right, totally missed that. There are no dependencies, so `patchShebang` should work just fine. I went with `patchPythonPrograms` anyways, just in case.

I don't think it belongs in `/bin`, since it is not a program a user should ever execute themselves (so it shouldn't be in `PATH`). The upstream installer script puts it in `~/.local/share`, so I've moved it to `/usr/share` instead.

Thanks for reviewing!

Reply:
I see, but then you should drop the `/usr` part by convention in nixpkgs

Reply:
Where do you get that convention from? A quick `grep` through nixpkgs doesn't show any instances of something being installed in `/share` and the FHS doesn't even mention it as a possibility.

Reply:
Unfortunately I don't think it's written down anywhere, but it makes sense to drop the /usr thing because we don't have a user/system separation anyways. I have no idea what kind of grep you're using though, but my ripgrep shows this just for $out/share: https://paste.infinisil.com/AzM8Qd7GaI.txt, it's very much a thing used throughout in nixpkgs, and NixOS also uses `/share/*` in `pathsToLink`.

Reply:
Oops I grepped for `/local/share` instead of `/share`.

I don't think it makes much sense, since I see no clear advantage in violating the standard here. Standard may come in handy in `buildUserFHSEnv` situations.

Anyway since its already a convention and I don't feel strongly either way, I've changed it.